lzy/allocator/src/main/java/ai/lzy/allocator/alloc/MountDynamicDiskAction.java

```diff
@@ -200,31 +198,39 @@ private StepResult prepareActiveMounts() {
         return StepResult.CONTINUE;
     }
 
-    private StepResult attachVolumeToPod() {
-        if (mountPod != null) {
-            return StepResult.ALREADY_DONE;
+    private StepResult getNextMountPodId() {
+        if (nextId != null) {
+            return StepResult.CONTINUE;
         }
-
-        var mountPodName = vm.instanceProperties().mountPodName();
-        if (mountPodName == null) {
-            log().error("{} Mount pod name is not found for vm", logPrefix());
-            fail(Status.FAILED_PRECONDITION.withDescription("Mount pod name is not found for vm " + vm.vmId()));
+        try {
+            nextId = withRetries(log(), () -> allocationContext.vmDao().getNextMountPodId(vm.vmId(), null));
+            Objects.requireNonNull(nextId, "nextId is null");
+        } catch (Exception e) {
+            log().error("{} Couldn't get next mount id for vm", logPrefix(), e);
+            fail(Status.CANCELLED.withDescription("Couldn't get next mount id for vm"));
             return StepResult.FINISH;
         }
+        return StepResult.CONTINUE;
+    }
+
+    private StepResult createNewMountPod() {
+        if (mountPod != null) {
+            return StepResult.ALREADY_DONE;
+        }
 
-        var mountPod = ClusterPod.of(dynamicMount.clusterId(), mountPodName);
         var dynamicMounts = new ArrayList<DynamicMount>(activeMounts.size() + 1);
         dynamicMounts.add(dynamicMount);
         dynamicMounts.addAll(activeMounts);
 
         log().info("{} Attaching mount {}", logPrefix(), dynamicMount.id());
         try {
-            this.mountPod = allocationContext.mountHolderManager().recreateWith(vm.spec(), mountPod, dynamicMounts);
+            mountPod = allocationContext.mountHolderManager().allocateMountHolder(vm.spec(), dynamicMounts,
+                nextId.toString());
         } catch (KubernetesClientException e) {
-            log().error("{} Couldn't attach mount {}", logPrefix(), dynamicMount.id(), e);
+            log().error("{} Couldn't create mount pod for {}", logPrefix(), dynamicMount.id(), e);
```

Comment:
is it possible to get `ALREADY_EXIST` here?

Reply:
`Already exist` should be resolved in manager. But as far as we create mount pod with consistent naming (see next_mount_pod_id), we can have weird situations. So we wait in `unmount action` for any pods to be deleted, and then we can execute next `mount action`. But this consistency between operations yet to be implemented